llrt_core/src/module_loader/resolver.rs

```diff
@@ -614,8 +614,11 @@ fn correct_extensions<'a>(x: String) -> Cow<'a, str> {
     if Path::new(&x).is_file() {
         return x.into();
     }
+
+    let index = if Path::new(&x).is_dir() { "/index" } else { "" };
+
```

Comment:
Create Path once and pull in metadata. is_dir and is_file each call metadata which adds extra cost. Better to reuse metadata and match on metadata.is_dir/is_file.